docs/calculations.mdx

```diff
@@ -0,0 +1,422 @@
+
+See [Consumer Devices](./consumer_devices)
+
+<ConsumerDeviceDefaults/>
+
+### Time in view for non-video ads
+
+Observations from various channels
+
+<TimeInViewDefaults/>
+
+### Ad platform request sizes by channel
+
+Observations from various channels
+
+<AdPlatformDefaults/>
+
+## Lookups from external sources
+
+### Carbon intensity by country, region, and UTC Date/Time
+
+Providers should clearly and publicly document:
+- Which data provider is used for carbon intensity data (for instance, WattTime or ElectricityMaps)
+- What carbon intensity metric is used (marginal vs average)
+
+`lookup_carbon_intensity_gco2e_per_kwh(country, region, utc_datetime)` should look up the carbon intensity in gCO2e based on 100-year global warming potential (GWP100) from a database at no less than hourly granularity.
+
+## Calculating Emissions
+
+### Creative Delivery - Data Transfer
+
+```
+creative_data_transfer_emissions_gco2pm =
+      creative_data_transfer_usage_emissions_gco2pm +
+      creative_data_transfer_embodied_emissions_gco2pm
+```
+
+For all channels other than CTV/BVOD, use the conventional model:
+
+```
+creative_data_transfer_bytes =
+      video_bytes +
+      audio_bytes +
+      image_bytes +
+      ad_format.video_player.size_bytes +
+      ad_format.other_asset_bytes
+
+creative_data_transfer_usage_emissions_gco2pm =
+      creative_data_transfer_bytes / 1024 x
+      default_usage_kwh_per_kb[network_type] x
+      lookup_carbon_intensity_gco2e_per_kwh(country, region, utc_datetime)
+
+creative_data_transfer_embodied_emissions_gco2pm =
+      creative_data_transfer_bytes / 1024 x
+      default_network_embodied_emissions_gco2e_per_kb[network_type]
+```
+
+If data transfer is provided in the delivery row, use it:
+
+```
+image_bytes = creative_image_data_transfer_bytes
+video_bytes = creative_video_data_transfer_bytes + creative_video_vast_bytes
+audio_bytes = creative_audio_data_transfer_bytes
+```
+
+Otherwise, calculate data transfer:
+
+```
+image_bytes = SUM(image_sizes[n].width x image_sizes[n].height) x 3 / default_image_compression_ratio
+
+audio_duration = creative_audio_duration_seconds ?? ad_format.audio_duration_seconds
+audio_bytes = audio_duration_seconds x default_audio_bitrate[device_type]
+
+default_bitrate =
+      ad_format.is_primary_experience ?
+      default_video_bitrate[device_type] :
+      default_video_bitrate_non_primary
+bitrate =
+      creative_video_bitrate_kbps ??
+      creative_video_size_bytes / 8 / creative_video_duration / 1024 ??
+      default_bitrate
+video_duration = creative_video_duration_seconds ?? ad_format.video_duration_seconds
+seconds_watched = creative_video_view_time ??
+                  creative_video_view_rate x video_duration ??
+                  video_duration
+buffer = ad_format.video_player.buffering_seconds ?? infinity
+seconds_streamed = MAX(seconds_watched + buffer, video_duration)
+loads = ad_format.video_player.lazy_loads ? views : impressions
```

Comment:
@pymble2073 should we add another conditional here for DOOH loads? and should the delivery row have loads? I guess we need this for images too